<?xml version="1.0"?>
<launch>
    <arg name="YOLO_topic_name" default="ishiyama_YOLO"/>
    <arg name="model_name" default="HV8"/>
    <arg name="radious" default="0.05"/>
    <arg name="timespan" default="4"/>
    <arg name="output_topic_name" default="/tanomimasu"/>
    <arg name="instance_number" default="0"/>
    
    <arg name="sim" default="true"/>
    <group if="$(arg sim)">
        <arg name="f_scale" default="0.5"/>
        <arg name="cx_scale" default="0.99"/>
        <arg name="cy_scale" default="0.9"/>
        <arg name="camera_topic_name" default="/photoneo_center/sensor/camera_info"/>
        <arg name="inputcloud_topic_name" default="cloud_without_segmented"/>
        <node name="Bridge" pkg="all_estimator" type="2D_to_3D_insnum_multi" output="screen">
            <param name="camera_topic_name" value="$(arg camera_topic_name)"/>
            <param name="YOLO_topic_name" value="$(arg YOLO_topic_name)"/>
            <param name="model_name" value="$(arg model_name)"/>
            <param name="f_scale" value="$(arg f_scale)"/>
            <param name="cx_scale" value="$(arg cx_scale)"/>
            <param name="cy_scale" value="$(arg cy_scale)"/>
            <param name="radious" value="$(arg radious)"/>
            <param name="timespan" value="$(arg timespan)"/>
            <param name="inputcloud_topic_name" value="$(arg inputcloud_topic_name)"/>
            <param name="output_topic_name" value="$(arg output_topic_name)"/>
            <param name="instance_number" value="$(arg instance_number)"/>
        </node>
    </group>

    <group unless="$(arg sim)">
        <arg name="f_scale" default="0.961"/>
        <arg name="cx_scale" default="1.184"/>
        <arg name="cy_scale" default="0.963"/>
        <arg name="camera_topic_name" default="/photoneo_center/camera_info"/>
        <arg name="inputcloud_topic_name" default="/photoneo_center/pointcloud"/>
        <node name="Bridge" pkg="all_estimator" type="2D_to_3D_insnum_multi" output="screen">
            <param name="camera_topic_name" value="$(arg camera_topic_name)"/>
            <param name="YOLO_topic_name" value="$(arg YOLO_topic_name)"/>
            <param name="model_name" value="$(arg model_name)"/>
            <param name="f_scale" value="$(arg f_scale)"/>
            <param name="cx_scale" value="$(arg cx_scale)"/>
            <param name="cy_scale" value="$(arg cy_scale)"/>
            <param name="radious" value="$(arg radious)"/>
            <param name="timespan" value="$(arg timespan)"/>
            <param name="inputcloud_topic_name" value="$(arg inputcloud_topic_name)"/>
            <param name="output_topic_name" value="$(arg output_topic_name)"/>
            <param name="instance_number" value="$(arg instance_number)"/>
        </node>
    </group>
    <!-- <arg name="f_scale" default="0.5"/>
    <arg name="cx_scale" default="0.99"/>
    <arg name="cy_scale" default="0.9"/> -->

    <!-- <arg name="f_scale" default="0.973"/>
    <arg name="cx_scale" default="1.03"/>
    <arg name="cy_scale" default="0.98"/> -->

    <!-- <arg name="f_scale" default="0.968"/>
    <arg name="cx_scale" default="1.01"/>
    <arg name="cy_scale" default="0.9"/> -->

</launch>